convert.py

```diff
@@ -727,6 +751,22 @@ def check_vocab_size(params: Params, vocab: Vocab) -> None:
             msg += f"  Most likely you are missing added_tokens.json (should be in {vocab.fname_tokenizer.parent})."
         raise Exception(msg)
 
+#### Mini Q8_0 quantization in Python
+QK8_0 = 32
+BLOCK_Q8_0 = np.dtype([('d', '<f2'), ('qs', 'i1', (QK8_0,))])
+def quantize_array_q8_0(arr):
+    assert arr.size % QK8_0 == 0 and arr.size != 0, f'Bad array size {arr.size}'
+    assert arr.dtype == np.float32, f'Bad array type {arr.dtype}'
+    n_blocks = arr.size // QK8_0
+    blocks = arr.reshape((n_blocks, QK8_0))
+    return np.fromiter(map(quantize_block_q8_0, blocks), count = n_blocks, dtype = BLOCK_Q8_0)
+
+def quantize_block_q8_0(blk, zero = np.float32(0), one = np.float32(1), onetwentyseven = np.float32(127), zero_chunk = (np.int8(0),) * QK8_0):
+    d = abs(blk).max() / onetwentyseven
+    if d == zero:
+        return (np.float16(d), zero_chunk)
+    return (np.float16(d), (blk * (one / d)).round())
```

Comment:
numpy is much more efficient if you give it bigger chunks. Feel free to pick better variable names.
```suggestion
    return np.fromiter(quantize_blocks_q8_0(blocks), count = n_blocks, dtype = BLOCK_Q8_0)
            
def quantize_blocks_q8_0(blks):
    d = abs(blks).max(axis=1, keepdims=True) / np.float32(127)
    d_f16 = np.float16(d) 
    qs = (blks * (np.float32(1) / d)).round()

    for i, (d, q) in enumerate(zip(d_f16, qs, strict=True)):
        yield d.item(), q
```

Reply:
Thanks for the feedback, that's definitely faster. Any idea of why the results are different though, and which would be correct?

Quantizing:

```plaintext
[-0.0062561  -0.00460815 -0.00753784 -0.00469971 -0.00100708  0.00210571
  0.01055908 -0.00692749  0.00285339  0.00491333 -0.00164795 -0.00028992
  0.00174713 -0.00141144  0.00842285  0.00111389  0.00994873 -0.00041771
 -0.00872803  0.00263977  0.01422119  0.01312256 -0.00836182 -0.0078125
 -0.01806641 -0.00939941  0.00256348 -0.00195312 -0.00119019  0.00415039
  0.0090332  -0.00485229]
```

Original:

    (0.0001422, [ -44,  -32,  -53,  -33,   -7,   15,   74,  -49,   20,   35,  -12,   -2,   12,  -10,   59,    8,   70,   -3,  -61,   19,  100,   92,  -59,  -55, -127,  -66,   18,  -14,   -8,   29,   64,  -34])

Yours:

    (0.0001422, [ -44,  -32,  -53,  -33,   -7,   15,   74,  -49,   20,   35,  -12,   -2,   12,  -10,   59,    8,   70,   -3,  -61,   19,  100,   92,  -59,  -55, -127,  -66,   18,  -14,   -8,   29,   63,  -34])

(The penultimate item in the original version is 64 while in yours it's 63.)

List version of the quantized values for easier pasting:

    [-0.0062561035, -0.0046081543, -0.007537842, -0.004699707, -0.0010070801, 0.002105713, 0.010559082, -0.0069274902, 0.0028533936, 0.00491333, -0.0016479492, -0.000289917, 0.0017471313, -0.001411438, 0.008422852, 0.0011138916, 0.0099487305, -0.00041770935, -0.008728027, 0.0026397705, 0.014221191, 0.013122559, -0.008361816, -0.0078125, -0.018066406, -0.009399414, 0.0025634766, -0.001953125, -0.0011901855, 0.0041503906, 0.009033203, -0.004852295]

Reply:
I can get your version to output 63 if I change `(blk * (one / d))` to `blk / d`. If you do the math in np.float64 you get 63.49996485, so I think my version is slightly more accurate.

Reply:
I can get behind that. Thanks! Performance is actually competitive with the C++ version now.